src/system/PredefinedProcedure.cpp

```diff
@@ -8,17 +8,36 @@
 #include <string>
 #include <utility>
 #include <vector>
+#include <iostream>
 
 using std::make_unique;
 using std::pair;
 using std::string;
 using std::vector;
+using std::cout;
 
 PredefinedProcedure::PredefinedProcedure(ProcKind kind, const string &name,
                                          const vector<pair<TypeNode *, bool>> &pairs, bool varargs, TypeNode *ret) :
-        ProcedureNode(make_unique<IdentDef>(name), nullptr), types_(), kind_(kind) {
+        ProcedureNode(make_unique<IdentDef>(name), nullptr), types_(), castIdx_(), kind_(kind) {
     auto type = overload(pairs, varargs, ret);
     this->setType(type);
+    castIdx_ = 0;
+    if (ret != nullptr) {
+        if (ret->kind() == TypeKind::TYPE) {
+            castIdx_ = -1;
+            int i = 1;
+            for (auto p : pairs) {
+                if (p.first->kind() == TypeKind::TYPE) {
+                    if (castIdx_ != -1) {
```

Comment:
Is this to guard against procedures that have two or more arguments of type type?

Reply:
Simplified and changed this to bool.